cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -76,76 +76,107 @@ let report = function (
 };
 
 /**
- * Requests a public certificate from AWS Certificate Manager, using DNS validation.
- * The hosted zone ID must refer to a **public** Route53-managed DNS zone that is authoritative
- * for the suffix of the certificate's Common Name (CN).  For example, if the CN is
- * `*.example.com`, the hosted zone ID must point to a Route 53 zone authoritative
- * for `example.com`.
+ * Requests a public certificate from AWS Certificate Manager, using DNS validation
+ * (see https://docs.aws.amazon.com/acm/latest/userguide/dns-validation.html).
+ * Specifically, it will do DNS validation in all the root, app, and env hosted zones in parallel.
+ * The root hosted zone is created when the user purchases example.com in route53 in their app account.
+ * We create the app hosted zone "app.example.com" when running "app init" part of the application stack.
+ * The env hosted zone "env.app.example.com" is created when running "env init" part of the env stack.
+ * Lastly, the function exits until the certificate is validated.
  *
  * @param {string} requestId the CloudFormation request ID
+ * @param {string} appName the name of the application
+ * @param {string} envName the name of the environment
  * @param {string} domainName the Common Name (CN) field for the requested certificate
- * @param {string} hostedZoneId the Route53 Hosted Zone ID
+ * @param {string} subjectAlternativeNames additional FQDNs to be included in the
+ * Subject Alternative Name extension of the requested certificate
+ * @param {string} envHostedZoneId the environment Route53 Hosted Zone ID
+ * @param {string} rootDnsRole the IAM role ARN that can manage domainName
+ * @param {string} isAliasEnabled whether alias is enabled
  * @returns {string} Validated certificate ARN
  */
 const requestCertificate = async function (
   requestId,
+  appName,
+  envName,
   domainName,
   subjectAlternativeNames,
-  hostedZoneId,
+  envHostedZoneId,
+  rootDnsRole,
+  isAliasEnabled,
   region
 ) {
   const crypto = require("crypto");
-  const [acm, route53] = clients(region);
+  const [acm, envRoute53, appRoute53] = clients(region, rootDnsRole);
+  subjectAlternativeNames =
+    isAliasEnabled === "false"
```

Comment:
lol 🤦 
nit: maybe we should create a new variable instead of modifying the input? 
I usually prefer doing this so that in the future if we have to modify this func again, we won't be confused why the variable is different than the passed value